src/java/org/apache/cassandra/cql3/statements/AlterTableStatement.java

```diff
@@ -276,6 +277,11 @@ public void validate(ClientState state)
             case DROP_COMPACT_STORAGE:
                 if (!meta.isCompactTable())
                     throw new InvalidRequestException("Cannot DROP COMPACT STORAGE on table without COMPACT STORAGE");
+                for (SSTableReader ssTableReader : Keyspace.open(keyspace()).getColumnFamilyStore(columnFamily()).getLiveSSTables())
```

Comment:
We could add a small TODO comment about the limitation to be addressed in CASSANDRA-15897.

Reply:
Added, thanks;